### PR TITLE
Fix flaky tests in hash join tests

### PR DESCRIPTION
**Pull request by xiaoxmeng**
The flaky tests in hash join test is because the recent changes that make hash build
operators to wait for probe side to finish. This caused the race condition that the task
to complete before all the drivers from the build side to close. For hash join, the probe
side pipeline is the output pipeline and a task will be terminated if the output pipeline
completes. This caused the build side stats is not reported and fails the tests. The
operator close (from driver close) aggregates the stats to task stats.

Run the tests internally for 200 iterations.

### CODE REVIEW

File: velox/exec/tests/HashJoinTest.cpp [hunk 1]
```diff
@@ -638,6 +638,20 @@ class HashJoinBuilder {
         memory::spillMemoryPool()->stats().peakBytes;
     TestScopedSpillInjection scopedSpillInjection(spillPct);
     auto task = builder.assertResults(referenceQuery_);
+    // Wait up to 5 seconds for all the task background activities to complete.
```
Comment (mbasmanova) [inline velox/exec/tests/HashJoinTest.cpp:641]: Would that also cause stats to be missing in production? There is no special wait like this there.

File: velox/exec/tests/HashJoinTest.cpp [hunk 2]
```diff
@@ -638,6 +638,20 @@ class HashJoinBuilder {
...
+    // Then we can collect the stats from all the operators.
+    uint64_t totalTaskWaitTimeUs{0};
+    while (task.use_count() != 1) {
```
Comment (mbasmanova) [inline velox/exec/tests/HashJoinTest.cpp:644]: Can we use something like Task::taskCompletionFuture(uint64_t maxWaitMicros) ?

Comment (xiaoxmeng) [reply velox/exec/tests/HashJoinTest.cpp:644]: It won't work as task terminate set the state to finish first and then close the drivers, and there will be a time race there.

**Review by mbasmanova — APPROVED** [review]
Accepting per offline discussion. Let's add a TODO to replace a sleep with waiting for some future and file a GitHub issue to provide Task API to wait for all background activities to finish.

**Pull request closed without merging**